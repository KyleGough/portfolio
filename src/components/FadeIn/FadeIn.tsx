import { clsx } from 'clsx';
import React, { useEffect, useState } from 'react';

import { useObserveElement } from '../../hooks/useObserveElement';

interface FadeInProps {
  className?: string;
  children: React.ReactNode;
}

export const FadeIn: React.FC<FadeInProps> = ({ className, children }) => {
  const [elementRef, isVisible] = useObserveElement<HTMLDivElement>({
    threshold: 0.5,
  });
  const [show, setShow] = useState(false);

  useEffect(() => {
    isVisible && setShow(true);
  }, [isVisible]);

  return (
    <div
      ref={elementRef}
      className={clsx(
        className,
        { 'opacity-100': show },
        { 'opacity-0': !show },
        'transition-opacity duration-1000'
      )}
    >
      {children}
    </div>
  );
};
