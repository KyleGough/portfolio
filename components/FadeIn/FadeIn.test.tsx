import { mockIntersectionObserver } from '@mocks/mockIntersectionObserver';
import { render, screen } from '@testing-library/react';
import React from 'react';

import { FadeIn } from './FadeIn';

describe('FadeIn component', () => {
  beforeEach(mockIntersectionObserver);
  it('renders', async () => {
    const { container } = render(<FadeIn>Test</FadeIn>);

    expect(screen.getByText('Test')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
