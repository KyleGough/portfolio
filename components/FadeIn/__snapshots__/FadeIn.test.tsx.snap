// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn component renders 1`] = `
<div>
  <div
    class="opacity-0 transition-opacity duration-1000"
  >
    Test
  </div>
</div>
`;
